kernel/actions/component/start.go

```diff
@@ -36,7 +36,11 @@ func (start *start) Execute(m *model.Model) error {
 		components := h.GetComponents(start.componentSpec)
 		for _, c := range components {
 			sshUsername := m.MustVariable("credentials", "ssh", "username").(string)
-			if err := internal.LaunchService(sshUsername, h.PublicIp, c.BinaryName, c.ConfigName); err != nil {
+			sshKeyPath := m.Variable("credentials", "ssh", "key_path").(string)
+
+			sshConfigFactory := internal.NewSshConfigFactoryImplWithKey(sshUsername, h.PublicIp, sshKeyPath)
```

Comment:
Couldn't we just pass the model to `sshConfigFactory` and let it look up the `sshUsername` and `sshKeyPath` variables? Would save duplicating those lookups every place they're used.

Reply:
Done.